src/app/layout/progressbar/RouteProgressBar.tsx

```diff
@@ -0,0 +1,58 @@
+import { useNProgress } from "@tanem/react-nprogress";
+import cx from "classnames";
+import React from "react";
+import { useNavigation, useLocation } from "react-router-dom";
+import css from "./RouteProgressBar.module.css";
+
+const LOADING_STATE = "loading";
+const ANIMATION_DURATION = 200;
+
+export const RouteProgress = () => {
+    const navigation = useNavigation();
+    const { key: locationKey } = useLocation();
+    const isLoading = navigation.state === LOADING_STATE;
+    // key is used to reset the state of the progress when location changes.
+    // Cannot use navigation.location.key directly because navigation.location
+    // will be undefined when navigation is not in a loading state, causing the animation
+    // to cancel early because the key will change at the same time as isLoading.
+    // Once navigation has finished, locationKey will be the same as previous (but now undefined)
+    // so fallback to that once navigation has finished.
+    const resetKey = navigation.location?.key || locationKey;
+
+    return <RouteProgressBar key={resetKey} isLoading={isLoading} />;
+};
+
+export const RouteProgressBar = ({ isLoading }) => {
+    const { isFinished, progress } = useNProgress({
+        animationDuration: ANIMATION_DURATION,
+        isAnimating: isLoading,
+    });
+
+    return (
+        <div
+            className={cx(css.progressBarWrapper, {
+                [css.finished]: isFinished,
+            })}
+            style={
+                {
+                    ["--animationDuration"]: `${ANIMATION_DURATION}ms`,
+                } as React.CSSProperties
+            }
+        >
+            <ProgressBar progress={progress} />
+        </div>
+    );
+};
+
+const ProgressBar = ({ progress }: { progress: number }) => {
+    return (
+        <div
+            className={css.progressBar}
+            style={
+                {
+                    ["--progress"]: `${(-1 + progress) * 100}%`,
```

Comment:
That's a good question - I just followed the example code for the css stuff. 
But seems to be because it seems to be a "trick" because it uses `width: 100%`. So it actually goes from `-100%` to `0%` . So eg. `0%` margin is 100% and `-10%` is visually 90% of progress.